apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/auth-methods/page-client.tsx

```diff
@@ -140,7 +118,207 @@ export default function PageClient() {
+              });
+            }}
+          />
+          <CardSubtitle className="mt-2">
+            SSO Providers
+          </CardSubtitle>
+
+          { enabledProviders.map(([, provider]) => provider)
+            .filter((provider): provider is AdminOAuthProviderConfig => !!provider).map(provider => {
+              return <div key={provider.id} className="flex h-10 items-center justify-between">
+                <div className="flex items-center gap-2">
+                  <div
+                    className="flex items-center justify-center w-12 h-12 rounded-md border border-gray-800"
+                    style={{ backgroundColor: BrandIcons.BRAND_COLORS[provider.id] ?? undefined }}
+                  >
+                    <BrandIcons.Mapping iconSize={24} provider={provider.id} />
+                  </div>
+                  <span className="text-sm font-semibold">{BrandIcons.toTitle(provider.id)}</span>
+                  {provider.type === 'shared' && <SimpleTooltip tooltip={SHARED_TOOLTIP}>
+                    <Badge variant="secondary">Shared keys</Badge>
+                  </SimpleTooltip>}
+                </div>
+
+                <OAuthActionCell config={provider} />
+              </div>;
+            }) }
+
+          <Button
+            className="mt-4"
+            onClick={() => {
+              setDisabledProvidersDialogOpen(true);
+            }}
+            variant="secondary"
+          >
+            <CirclePlus size={16}/>
+            <span className="ml-2">Add SSO providers</span>
+          </Button>
+          <DisabledProvidersDialog
+            open={disabledProvidersDialogOpen}
+            onOpenChange={(x) => {
+              setDisabledProvidersDialogOpen(x);
+            }}
+          />
+        </SettingCard>
+        <SettingCard className="hidden lg:flex">
+          <div className="self-stretch py-4 px-4 min-w-[400px] items-center">
+            <div className="w-full">
+              <BrowserFrame url="your-website.com/signin">
+                <div className="flex flex-col items-center justify-center min-h-[400px]">
+                  <div className='w-full sm:max-w-xs m-auto scale-90 pointer-events-none' inert=''>
+                    {/* a transparent cover that prevents the card from being clicked, even when pointer-events is overridden */}
+                    <div className="absolute inset-0 bg-transparent z-10"></div>
+                    <AuthPage
+                      type="sign-in"
+                      mockProject={{
+                        config: {
+                          ...project.config,
+                          oauthProviders: enabledProviders
+                            .map(([, provider]) => provider)
+                            .filter((provider): provider is AdminOAuthProviderConfig => !!provider),
+                        },
+                      }}
+                    />
+                  </div>
+                </div>
+              </BrowserFrame>
```

Comment:
can you extract this whole thing into a component and then use it for both the Create Project page and here?

Reply:
Will extract it now, then open a new PR with it